powermon/commands/command.py

```diff
@@ -26,7 +28,7 @@ def __str__(self):
         return f"Command: {self.name=} {self.full_command=}, {self.type=}, [{_outs=}], {last_run=}, {next_run=}, {str(self.trigger)}, {self.command_defn=}"
 
     @classmethod
-    def fromConfig(cls, config=None):
+    def fromConfig(cls, config=None, mqtt_broker=None) -> "Command":
```

Comment:
would really prefer fromConfig to be only config processing
passing mqtt object feels like a future problem (even tho i use similar in places), one problem in the mppsolar code is that it is hard to add extra/different approaches - eg when someone added DB storage a bunch of function definitions needed to extend to include the DB info/object
maybe device (or some more generic object) should wrap messaging etc - so device.publish() can do mqtt but could also do some other type of messaging....

Reply:
Good point, it should only be config. Maybe a builder pattern can help here, the fromConfig can return a builder that we can use to add more information that doesn't come directly from the config.

Reply:
I shouldn't be the one resolving it